### PR TITLE
gomod: update zoekt for List performance improvements (#61397)

### PR DESCRIPTION
**Pull request by michaellzc**
https://github.com/sourcegraph/zoekt/compare/f2577c25dc61...931974dd953e

- 06de7adb72 shards: use selectRepoSet in List
- 931974dd95 shards: respect scheduler and use smarter synchronization for List

Test Plan: tested in zoekt repo and this CI

(cherry picked from commit b98f333ba5f16f7ad410ba6335758f3aa3f4a4b0)

backport https://github.com/sourcegraph/sourcegraph/pull/61397

## Test plan

CI

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/sourcegraph/sourcegraph
 
-go 1.21
```
Comment (eseliger) [inline go.mod:3]: this feels risky, was that required?

Comment (michaellzc) [reply go.mod:3]: I hope not. let me revert this

Comment (mmanela) [reply go.mod]: Ugh this is the rub I guess, we have changes backporting that may already take advantage of 1.22. (like the loop capture change), so not sure the impact of backporting and leaving old go version @eseliger

Comment (michaellzc) [reply go.mod]: reverted, 🤞 CI passes

Comment (eseliger) [reply go.mod]: sourcegraph/zoekt asks for go 1.22, maybe it actually is required :/

Comment (michaellzc) [reply go.mod]: > sourcegraph/zoekt asks for go 1.22, maybe it actually is required :/

we can maintain a 1.21 branch of zoekt if needed. let's see

Comment (mmanela) [reply go.mod]: but this PR is updating to a new build of zoekt

Comment (eseliger) [reply go.mod]: > Ugh this is the rub I guess, we have changes backporting that may already take advantage of 1.22. (like the loop capture change)

Good point, that might actually be risky. But just randomly bumping the go version in an emergency patch also feels risky :/

**Comment by keegancsmith** [comment]
I will create a new backport for this that is far less risky. ie doesn't change go versions and only includes our targetted changes instead of all changes to zoekt main since 5.3

**Pull request closed without merging**